src/store/ethereum/wallets/binance.ts

```diff
@@ -69,7 +72,25 @@ async function changeAccounts(accounts: string[]) {
     ethereumModule.state.address.toLowerCase() !== accounts[0]) {
     // Remove any reference to past withdrawals as it is bound to a specific address
     localStorage.removeItem("lastWithdrawTime")
-    ethereumModule.state.metamaskChangeAlert = true;
+    ethereumModule.state.metamaskChangeAlert = true
     // (window as any).BinanceChain.removeAllListeners() // NOTE: not implemented by Binance Wallet
   }
 }
+
+/**
+ * https://github.com/loomnetwork/dashboard/issues/1421
+ * NOTE: based on ethers 4.0.47!
+ */
+function patchSigner(signer: ethers.Signer) {
+  const patch = async (message: ethers.utils.Arrayish) => {
+    const data = ((typeof (message) === "string") ? ethers.utils.toUtf8Bytes(message) : message)
+    const address = await signer.getAddress()
+    // @ts-ignore
+    return signer.provider._web3Provider.bnbSign(address, ethers.utils.hexlify(data))
+      .then((result) => result.signature)
+      .catch((result) => { throw new Error("Biance wallet: " + result.error) })
```

Comment:
The braces are redundant here I think. Also typo.
```js
throw new Error("Binance wallet bnbSign failed: " + result.error)
```

Reply:
no braces is a shorthand for return. Can't have `return throw`